### ISSUE TITLE
import/no-unused-modules with `unusedExports: true` flag destructured array exports as undefined

### ISSUE THREAD

**Issue by astorije** [benmosher/eslint-plugin-import#1683]
Given the following code:

```tsx
export const [foobar] = foobarFactory();
```

`import/no-unused-modules` with the `unusedExports: true` option will report:

```
error  exported declaration 'undefined' not used within other modules  import/no-unused-modules
```

This is true whether or not `foobar` is being imported anywhere else.

**Comment by ljharb** [comment]
hmm, this test case passes on latest master. Perhaps it's already been fixed for the next release?
